x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ldap/PoolingSessionFactory.java

```diff
@@ -104,17 +105,28 @@ private SimpleBindRequest buildBindRequest(Settings settings) {
             return new SimpleBindRequest();
         } else {
             if (bindPassword == null) {
-                deprecationLogger.critical(
-                    DeprecationCategory.SECURITY,
-                    "bind_dn_set_without_password",
-                    "[{}] is set but no bind password is specified. Without a corresponding bind password, "
-                        + "all {} realm authentication will fail. Specify a bind password via [{}] or [{}]. "
-                        + "In the next major release, nodes with incomplete bind credentials will fail to start.",
-                    RealmSettings.getFullSettingKey(config, PoolingSessionFactorySettings.BIND_DN),
-                    config.type(),
-                    RealmSettings.getFullSettingKey(config, SECURE_BIND_PASSWORD),
-                    RealmSettings.getFullSettingKey(config, LEGACY_BIND_PASSWORD)
-                );
+                if (failOnMissingBindPassword) {
+                    throw new SettingsException(
+                        "[{}] is set but no bind password is specified. Without a corresponding bind password, "
+                            + "all {} realm authentication will fail. Specify a bind password via [{}] or [{}].",
+                        RealmSettings.getFullSettingKey(config, PoolingSessionFactorySettings.BIND_DN),
+                        config.type(),
+                        RealmSettings.getFullSettingKey(config, SECURE_BIND_PASSWORD),
+                        RealmSettings.getFullSettingKey(config, LEGACY_BIND_PASSWORD)
```

Comment:
The only thing I'm not sure is if we want to advertise using a deprecated non-secure setting here.

Reply:
Thanks for the suggestions. After thinking about this a bit more, I think the correct option is **b)**, given that `LEGACY_BIND_PASSWORD` is a node scoped, non-secure setting that cannot be dynamically changed via reload API. Unit tests were suggesting that both legacy and secure settings can be reloaded, which is wrong. So, I've changed them.